### PR TITLE
sql: add logictest for CCL

### PR DESCRIPTION
**Pull request by mberhault**
Simple tests for CCL-only statements in OSS logic test.
Brain-dead test in CCL logictest just to show it works.

Release note: None

### CODE REVIEW

**Comment by mberhault** [comment]
@benesch: since you said you'd worked on this before.

File: pkg/sql/logictest/logic_runner_test.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2015 The Cockroach Authors.
```
Comment (mberhault) [inline pkg/sql/logictest/logic_runner_test.go]: I wanted to name this `logic_test.go`, but then it doesn't give you a nice "moved + a few changes" in the diff for the original `logic_test.go`. We can rename it later.

File: Makefile [hunk 2]
```diff
@@ -766,6 +773,14 @@ testlogic: TESTFLAGS := -test.v $(if $(FILES),-show-sql)
 testlogic: bin/logictest.test
 	cd pkg/sql/logictest && logictest.test -test.run "$(TESTS)" -test.timeout $(TESTTIMEOUT) $(TESTFLAGS)
 
+# Run make testccllogic to run all of the CCL logic tests. Specify test files to run
+# with make testccllogic FILES="foo bar".
+testccllogic: ## Run SQL Logic Tests.
```
Comment (mberhault) [inline Makefile]: `testlogicccl` would fit better with our naming convention, but it awkward to type/say/read.

File: Makefile [hunk 3]
```diff
@@ -766,6 +773,14 @@ testlogic: TESTFLAGS := -test.v $(if $(FILES),-show-sql)
...
+testccllogic: TESTS := $(if $(FILES),TestLogic$$//^$(subst $(space),$$|^,$(FILES))$$$(if $(SUBTESTS),/$(SUBTESTS)),TestLogic)
+testccllogic: TESTFLAGS := -test.v $(if $(FILES),-show-sql)
```
Comment (benesch) [inline Makefile]: Ditto here. Can you share at least these two recipe-specific variables with `testlogic`?

```make
testlogic testccllogic: TESTS := $(if $(FILES),TestLogic$$//^$(subst $(space),$$|^,$(FILES))$$$(if $(SUBTESTS),/$(SUBTESTS)),TestLogic)
testlogic testccllogic: TESTFLAGS := -test.v $(if $(FILES),-show-sql)
```

File: pkg/sql/logictest/testdata/logic_test/restore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# LogicTest: default
+
+# Restore is CCL only.
+statement error pq: unknown statement type: \*tree\.Restore
+RESTORE foo FROM "bar"
```
Comment (benesch) [inline pkg/sql/logictest/testdata/logic_test/restore]: Can you add `BACKUP` just for completeness's sake? (And perhaps call this file "ccl" or "enterprise" or something more generic.)

File: Makefile [hunk 5]
```diff
@@ -748,12 +749,18 @@ bin/logictest.test: main.go $(shell $(FIND_RELEVANT) ! -name 'zcgo_flags.go' -na
 	$(MAKE) gotestdashi GOFLAGS='$(GOFLAGS)' TAGS='$(TAGS)' LINKFLAGS='$(LINKFLAGS)' PKG='$(PKG)'
 	$(XGO) test $(GOFLAGS) -tags '$(TAGS)' -ldflags '$(LINKFLAGS)' -c -o bin/logictest.test $(PKG)
 
+bin/logictestccl.test: PKG := ./pkg/ccl/sqlccl/logictestccl
+bin/logictestccl.test: main.go $(shell $(FIND_RELEVANT) ! -name 'zcgo_flags.go' -name '*.go')
+	$(MAKE) gotestdashi GOFLAGS='$(GOFLAGS)' TAGS='$(TAGS)' LINKFLAGS='$(LINKFLAGS)' PKG='$(PKG)'
+	$(XGO) test $(GOFLAGS) -tags '$(TAGS)' -ldflags '$(LINKFLAGS)' -c -o bin/logictestccl.test $(PKG)
```
Comment (benesch) [inline Makefile]: Mind sharing the existing rule instead?

```make
bin/logictest.test: PKG := ./pkg/sql/logictest
bin/logictestccl.test: PKG := ./pkg/ccl/sqlccl/logictestccl

bin/logictest.test bin/logictestccl.test: main.go $(shell $(FIND_RELEVANT) ! -name 'zcgo_flags.go' -name '*.go')
	$(MAKE) gotestdashi GOFLAGS='$(GOFLAGS)' TAGS='$(TAGS)' LINKFLAGS='$(LINKFLAGS)' PKG='$(PKG)'
	$(XGO) test $(GOFLAGS) -tags '$(TAGS)' -ldflags '$(LINKFLAGS)' -c -o $@ $(PKG)
```

**Comment by benesch** [comment]
In `logictestccl.MainTest`, I would just call `TestingEnableEnterprise`, like `sqlccl` does.

**Review by benesch — APPROVED** [review]
Nice!

File: Makefile (hunk 5 above)
Comment (mberhault) [reply Makefile]: Good point. Done.

File: Makefile (hunk 3 above)
Comment (mberhault) [reply Makefile]: Done.

File: pkg/sql/logictest/testdata/logic_test/restore (hunk 4 above)
Comment (mberhault) [reply pkg/sql/logictest/testdata/logic_test/restore]: Done. renamed `ccl`.

**Comment by mberhault** [comment]
Enabled test license, thanks for the pointer, I figured something like that existed I just hadn't looked yet.
I'll go ahead and rename `logic_runner_test` as well.

**Pull request merged**